lib/modules/manager/bundler/extract.spec.ts

```diff
@@ -29,6 +30,18 @@ const sourceBlockWithNewLinesGemfile = Fixtures.get(
 const sourceBlockWithGroupsGemfile = Fixtures.get(
   'Gemfile.sourceBlockWithGroups',
 );
+const sourceVariableGemfile = codeBlock`
+source "https://rubygems.org"
+ruby '~> 1.5.3'
+foo = 'https://gems.foo.com'
+bar = 'https://gems.bar.com'
+
+source foo
```

Comment:
> If we place dependency right after this statement, it should have https://gems.foo.com as the registry?

that's my understanding too. At least, here the registry will be added to the global list, so it would be tried (if I got this correctly).

Reply:
If this works, please add test for this case too